src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -205,58 +206,49 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
         }));
     }
 
-    private async Task WatchDashboardLogsAsync()
+    private async Task WatchDashboardLogsAsync(CancellationToken cancellationToken)
     {
-        async Task<string?> GetDashboardResourceIdentifier()
-        {
-            string? dashboardResourceId = null;
-
-            try
-            {
-                var timeout = Debugger.IsAttached ? Timeout.InfiniteTimeSpan : dashboardOptions.Value.DashboardStartupTimeout;
+        var loggerCache = new ConcurrentDictionary<string, ILogger>(StringComparer.Ordinal);
+        var defaultDashboardLogger = loggerFactory.CreateLogger("Aspire.Hosting.Dashboard");
 
-                // The dashboard resource isn't immediately available so watch for it.
-                // Wait for dashboard to startup and be reported before giving up.
-                using var cts =  new CancellationTokenSource(timeout);
-                using var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cts.Token, _shutdownCts.Token);
+        var dashboardResourceTasks = new Dictionary<string, Task>();
```

Comment:
```suggestion
        var dashboardResourceTasks = new Dictionary<string, Task>(StringComparers.ResourceName);
```

Reply:
The notifier is case sensitive currently.